framework/src/postprocessors/ElementAverageValue.C

```diff
@@ -43,9 +43,7 @@ Real
 ElementAverageValue::getValue()
 {
   Real integral = ElementIntegralVariablePostprocessor::getValue();
-
   gatherSum(_volume);
-
```

Comment:
what about this one?

Reply:
Ah, these are the result of having to test backwards to see what broke. I moved every gatherSum() call into a finalize that made sense, and then went back and re-added the lines after testing revealed downstream problems. So because of this it the final git diff just looks like I removed some lines, haha

Reply:
If it's a big deal I can change them back to their original whitespace.

Reply:
Do you know why moving this to `finalize` caused problems?

Reply:
The deleted new lines are not a problem. That's nice. I just would like to understand why moving `gatherSum` from `getValue` to `finalize` caused problems. In an ideal world the should be interchangeable, so when they're not, I would like to understand why, if possible.

I know there are `gatherSum` often in base class `getValue` methods, like in `ElementIntegralPostprocessor`... don't know if those are playing a role here (would be nice to get that one moved as well)

Reply:
I'm not entirely sure what changed exactly internally, but the relative diff on some tests would always change to a pretty high number (according to casey, anyway). My guess is the sum isn't being passed through correctly.

Reply:
I could investigate further if desired

Reply:
Worked on it this week under Cody's direction. After some debugging and testing issues should be sorted out. Just waiting on a Bison PR to merge with downstream changes so I can restart the internal app tests and it should be good to go.